lib/srv/desktop/rdp/rdpclient/src/rdpdr/filesystem.rs

```diff
@@ -352,6 +355,109 @@ impl FilesystemBackend {
         self.send_drive_query_dir_response(rdp_req.device_io_request, NtStatus::UNSUCCESSFUL, None)
     }
 
+    fn handle_query_volume_req(
+        &mut self,
+        rdp_req: efs::ServerDriveQueryVolumeInformationRequest,
+    ) -> PduResult<()> {
+        if let Some(dir) = self.file_cache.get(rdp_req.device_io_request.file_id) {
+            let buffer: Option<efs::FileSystemInformationClass> = match rdp_req.fs_info_class_lvl {
+                efs::FileSystemInformationClassLevel::FILE_FS_VOLUME_INFORMATION => {
+                    Some(
+                        efs::FileFsVolumeInformation {
+                            volume_creation_time: cast_length!(
+                                "FilesystemBackend::handle_query_volume_req",
+                                "dir.fso.last_modified",
+                                dir.fso.last_modified
+                            )?,
+                            // Not sure why the `& 0xffff` is necessary, just copying FreeRDP
+                            // https://github.com/FreeRDP/FreeRDP/blob/511444a65e7aa2f537c5e531fa68157a50c1bd4d/channels/drive/client/drive_main.c#L492
+                            // u32::MAX is given due to the fact that FreeRDP uses it as a fallback:
+                            // https://github.com/FreeRDP/FreeRDP/blob/511444a65e7aa2f537c5e531fa68157a50c1bd4d/winpr/libwinpr/file/file.c#L1018-L1021
+                            volume_serial_number: u32::MAX & 0xffff,
```

Comment:
From http://msdn.microsoft.com/en-us/library/cc232108.aspx:
> VolumeSerialNumber (4 bytes): A 32-bit unsigned integer that contains the serial number of the volume. The serial number is an opaque value generated by the file system at format time, and is not necessarily related to any hardware serial number for the device on which the file system is located. No specific format or content of this field is required for protocol interoperation. This value is not required to be unique.

So you can use any value you want, `0xfff` is as good as any but `u32::MAX &` is redundant

Reply:
11f4d6aff990b0da0cf420d50bff76d45b7f39f7

Reply:
11f4d6aff990b0da0cf420d50bff76d45b7f39f7